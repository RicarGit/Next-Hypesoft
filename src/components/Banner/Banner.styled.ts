import styled from "styled-components"

export const BannerSection = styled.section`
  width: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;
  margin-top: 166px;
  padding: 64px 10px;
  gap: 32px;
  background-image: url('banner-shapes.svg');
  background-repeat: no-repeat;
  background-size: cover;
  background-position: right;
  
  button {
    width: auto;
  }

  @media (max-width: 530px) {
    padding: 20px 10px;
    text-align: center;
  }
`

export const BannerTitle = styled.h2`
  font-size: 32px;
  font-weight: 400;
  line-height: 48px;
  color: ${({ theme }) => theme.colors.white};

  span {
    color: ${({ theme }) => theme.colors.secondary};
  }

  @media (max-width: 530px) {
    font-size: 26px;
    line-height: 36px;
  }
`