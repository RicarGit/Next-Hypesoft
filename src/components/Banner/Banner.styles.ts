import styled from "styled-components"

export const BannerSection = styled.section`
  width: 100%;
  position: absolute;
  top: 4126px;
  padding: 64px;
`

export const Container = styled.div`
  width: 1320px;
  height: 184px;
  margin: 0 auto;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  gap: 32px;

  button {
    width: auto;
  }
`

export const BannerTitle = styled.h2`
  font-size: 32px;
  font-weight: 400;
  line-height: 48px;
  color: ${({ theme }) => theme.colors.white};

  span {
    color: ${({ theme }) => theme.colors.secondary};
  }
`