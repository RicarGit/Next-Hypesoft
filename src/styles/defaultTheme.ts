import { DefaultTheme } from "styled-components"

const defaultTheme: DefaultTheme = {
  colors: {
    primary: '#242424',
    secondary: '#ED1CA6',
    sec: '#7D00FF',
    white: '#FFFFFF',
    gray: {
      100: '#DFE3EB',
      500: '#A5A5A5',
    },
  },
  font: {
    family: 'Inter',
    weigths: {
      regular: 400,
      medium: 500,
      semiBold: 600,
      bold: 700
    },
  },
}

export default defaultTheme