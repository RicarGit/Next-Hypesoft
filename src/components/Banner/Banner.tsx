import * as S from './Banner.styles'

import Image from 'next/image'
import bannerShapes from './shapes.svg'

import Button from 'shared/Button'

export const Banner = () => {
  return (
    <S.BannerSection>
      <S.Container>

        <S.BannerShapesWrapper>
          <Image src={bannerShapes} alt='banner' />
        </S.BannerShapesWrapper>

        <S.BannerTitle>
          Got a project in mind? Let's make <br />
          something awesome <span>together</span>.
        </S.BannerTitle>

        <Button>Hire Me</Button>
      </S.Container>
    </S.BannerSection>
  )
}