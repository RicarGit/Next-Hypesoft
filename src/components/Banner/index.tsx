import * as S from './styles'

import Image from 'next/image'
import bannerShapes from './shapes.svg'

import Button from 'components/shared/Button'

const Banner = () => {
  return (
    <S.BannerSection>
      <S.Container>

        <S.BannerShapesWrapper>
          <Image src={bannerShapes} alt='banner' />
        </S.BannerShapesWrapper>

        <S.BannerTitle>
          Got a project in mind? Let's make <br />
          something awesome <span>together</span>.
        </S.BannerTitle>

        <Button>Hire Me</Button>
      </S.Container>
    </S.BannerSection>
  )
}

export default Banner