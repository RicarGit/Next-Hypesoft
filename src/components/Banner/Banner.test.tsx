import { render } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'
import defaultTheme from 'styles/defaultTheme'
import Banner from '.'

describe('Banner', () => {
  it('should match snapshot', () => {
    const { container } = render(
      <ThemeProvider theme={defaultTheme}>
        <Banner />
      </ThemeProvider>
    )

    expect(container.firstChild).toMatchSnapshot()
  })
})