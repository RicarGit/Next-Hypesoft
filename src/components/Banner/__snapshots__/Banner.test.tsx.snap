// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner should match snapshot 1`] = `
<section
  class="sc-bczRLJ coFFbx"
>
  <div
    class="sc-gsnTZi ezMaWa"
  >
    <div
      class="sc-dkzDqf fpcBRk"
    >
      <span
        style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
      >
        <span
          style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2724%27%20height=%2724%27/%3e"
            style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
          />
        </span>
        <img
          alt="banner"
          data-nimg="intrinsic"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </span>
    </div>
    <h2
      class="sc-hKMtZM gtxSEF"
    >
      Got a project in mind? Let's make 
      <br />
      something awesome 
      <span>
        together
      </span>
      .
    </h2>
    <button
      class="sc-eCYdqJ JGqaN"
    >
      <p
        class="sc-jSMfEi hICwZx"
      >
        Hire Me
      </p>
      <span
        style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
      >
        <span
          style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2724%27%20height=%2724%27/%3e"
            style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
          />
        </span>
        <img
          alt="white arrow"
          data-nimg="intrinsic"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </span>
    </button>
  </div>
</section>
`;
